app/main.py

```diff
@@ -1,2 +1,94 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Any, Hashable
+
+
+@dataclass
+class Node:
+    key: Hashable
+    hash_: int
+    value: Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.threshold = int(self.capacity * (2 / 3)) + 1
+        self.length = 0
+        self.hash_table: list = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+
+        if self.length == self.threshold:
+            self.resize()
+
+        hash_ = hash(key)
+        index = hash_ % self.capacity
+        node = Node(key, hash_, value)
+
+        while self.hash_table[index] is not None:
+            if self.hash_table[index].key == key:
+                self.hash_table[index] = node
+                return
+            index = (index + 1) % self.capacity
+
+        self.hash_table[index] = node
+        self.length += 1
+
+    def __getitem__(self, key: Any) -> Any:
+        index = hash(key) % self.capacity
+        while self.hash_table[index] is not None:
+            if self.hash_table[index].key == key:
+                return self.hash_table[index].value
+            index = (index + 1) % self.capacity
+        raise KeyError
+
+    def resize(self) -> None:
+        old_hash_table = [item for item in self.hash_table if item is not None]
+        self.__init__(self.capacity * 2)
+        for item in old_hash_table:
+            self.__setitem__(item.key, item.value)
+
+    def clear(self) -> None:
+        self.__init__()
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = hash(key) % self.capacity
```

Comment:
To make the code more DRY and improve readability, you can create a separate function for getting the index and then use it where needed